polymetis/polymetis/include/polymetis/polymetis_server.hpp

```diff
@@ -12,6 +12,9 @@
 #include <unistd.h>
 #include <vector>
 
+#include <sys/ipc.h>
```

Comment:
These are SysV shared memory. I would avoid it in favor of posix shared memory, a.k.a. tmpfs, a.k.a. /dev/shm

Edit: I don't see you use shmget, shmat, ... I think you're just using boost, which I believe calls shm_open (the posix one) under the hood.